omegaconf/listconfig.py

```diff
@@ -103,8 +98,10 @@ def _validate_set(self, key: Any, value: Any) -> None:
                         "$FULL_KEY is not optional and cannot be assigned None"
                     )
 
-        target_type = self._metadata.element_type
+        is_optional, target_type = _resolve_optional(self._metadata.element_type)
         value_type = OmegaConf.get_type(value)
+        if is_optional and value_type is None:
+            return
         if is_structured_config(target_type):
             if (
                 target_type is not None
```

Comment:
I wasn't sure how to test this particular diff. The tests still pass even if this diff is not included in the PR.

Reply:
Looking at it, this seems to be because the validation error is raised anyway when trying to set the value, even if `_validate_set()` didn't catch it.

You could add a test that explicitly tests `_validate_set()` in a situation like below (it raises an exception if and only if this diff is included):
```python
@dataclass
class A:
    x: int = 0

cfg = ListConfig([], element_type=Optional[A])
cfg._validate_set(0, 3)
```

Reply:
That works, thanks! I added some tests in https://github.com/omry/omegaconf/pull/749/commits/2bc4023eaac93690fbc22f9a1134def4c30ee06b, and additionally (with the help of TDD) I found another edge case where ValidationError should be raised (`value_type==None` and `element_type` is not optional).

Reply:
Sounds good, resolving.